### PR TITLE
Update CONTRIBUTING guide and Kafka versions

### PR DESCRIPTION
**Pull request by embano1**
Description
-----------

- [x] Add testing instructions to `CONTRIBUTING ` guide
- [x] Update Kafka version in `docker-compose` files

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,6 +40,138 @@ GitHub provides additional document on [forking a repository](https://help.githu
...
+
+To create the (shaded) JAR(s) run:
+
+```bash
+mvn clean package -Drevision=$(git describe --tags --always)
+```
+
+#### Integration Tests with Docker Compose
+
+To execute the integration tests (requires [Docker](https://docker.com/)) run:
```
Comment (baldawar) [inline CONTRIBUTING.md:77]: Is there an alternative we can offer for organizations greater than 250 who do not have a license https://www.docker.com/pricing/faq/ 

> Docker Desktop requires a paid, per-user subscription for organizations with more than 250 employees or more than $10 million in annual revenue per our [terms of service](https://www.docker.com/legal/docker-terms-service/).

Comment (embano1) [reply CONTRIBUTING.md:77]: Good catch! Will ammend!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -40,6 +40,138 @@ GitHub provides additional document on [forking a repository](https://help.githu
...
+
+```bash
+export KAFKA_VERSION=3.5.0
+export COMPOSE_FILE=e2e/docker_compose.yaml
+mvn clean verify -Drevision=$(git describe --tags --always)
```
Comment (baldawar) [inline CONTRIBUTING.md:82]: Just a heads up, this line triggered a failed build 

```
[main] INFO tc.docker/compose:1.29.2 - Docker Compose has finished running
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 245.1 s <<< FAILURE! -- in software.amazon.event.kafkaconnector.EventBridgeSinkConnectorIT
[ERROR] software.amazon.event.kafkaconnector.EventBridgeSinkConnectorIT -- Time elapsed: 245.1 s <<< FAILURE!
org.opentest4j.AssertionFailedError: failed to run docker compose environment: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [57872, 8083] should be listening)
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:38)
	at org.junit.jupiter.api.Assertions.fail(Assertions.java:135)
	at software.amazon.event.kafkaconnector.EventBridgeSinkConnectorIT.startDockerComposeEnvironment(EventBridgeSinkConnectorIT.java:115)
	at software.amazon.event.kafkaconnector.EventBridgeSinkConnectorIT.setup(EventBridgeSinkConnectorIT.java:88)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptLifecycleMethod(TimeoutExtension.java:128)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptBeforeAllMethod(TimeoutExtension.java:70)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllMethods$13(ClassBasedTestDescriptor.java:411)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllMethods(ClassBasedTestDescriptor.java:409)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:215)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   EventBridgeSinkConnectorIT.setup:88->startDockerComposeEnvironment:115 failed to run docker compose environment: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [57872, 8083] should be listening)
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-failsafe-plugin:3.1.2:verify (default) @ kafka-eventbridge-sink ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:22 min
[INFO] Finished at: 2023-07-06T08:23:07-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.1.2:verify (default) on project kafka-eventbridge-sink: There are test failures.
[ERROR] 
[ERROR] Please refer to /Volumes/Unix/workspaces/kafka-connector/eventbridge-kafka-connector/target/failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -40,6 +40,138 @@ GitHub provides additional document on [forking a repository](https://help.githu
...
+locally e.g., to create the local environment:
+
+```bash
+# create the JAR which will be loaded into Kafka Connect (Docker Volume)
+mvn clean package -Drevision=$(git describe --tags --always)
+
+# move to the e2e directory
+cd e2e
+
+# set credentials used by the connector to send events to EventBridge (or assume a corresponding role)
```
Comment (baldawar) [inline CONTRIBUTING.md:108]: This is only necessary if credentials aren't set already through other means, right? If so, we should recommend skipping when unnecessary.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -40,6 +40,138 @@ GitHub provides additional document on [forking a repository](https://help.githu
...
+e2e-connect-1     | [2023-07-04 09:54:55,825] TRACE [eventbridge-e2e|task-0] Sending request to EventBridge: PutEventsRequest(Entries=[PutEventsRequestEntry(Source=kafka-connect.eventbridge-e2e-connector, Resources=[], DetailType=kafka-connect-eventbridge-e2e, Detail={"topic":"eventbridge-e2e","partition":0,"offset":1,"timestamp":1688464495808,"timestampType":"CreateTime","headers":[],"key":null,"value":{"hello":"world"}}, EventBusName=arn:aws:events:us-east-1:1234567890:event-bus/eventbridge-e2e)]) (software.amazon.event.kafkaconnector.EventBridgeWriter:152)
+e2e-connect-1     | [2023-07-04 09:54:56,100] TRACE [eventbridge-e2e|task-0] putEvents response: [PutEventsResultEntry(EventId=29725ba0-c013-9457-0be2-51ba26708d3d)] (software.amazon.event.kafkaconnector.EventBridgeWriter:154)
+e2e-connect-1     | [2023-07-04 09:54:56,301] TRACE [eventbridge-e2e|task-0] putItems call completed: start=2023-07-04T09:54:55.824953210Z completion=2023-07-04T09:54:56.301049627Z durationMillis=476 attempts=1 maxRetries=2 (software.amazon.event.kafkaconnector.EventBridgeSinkTask:105)
+```
+
```
Comment (baldawar) [inline CONTRIBUTING.md:168]: can add here that the output event from on your bus should look  like 

```json
{
    "version": "0",
    "id": "467efba4-6a53-d275-a52c-416a0c22abe4",
    "detail-type": "kafka-connect-eventbridge-e2e",
    "source": "kafka-connect.eventbridge-e2e-connector",
    "account": "123456789012",
    "time": "2023-07-06T16:02:19Z",
    "region": "us-east-1",
    "resources": [],
    "detail": {
        "topic": "eventbridge-e2e",
        "partition": 0,
        "offset": 0,
        "timestamp": 1688659338101,
        "timestampType": "CreateTime",
        "headers": [],
        "key": null,
        "value": {
            "hello": "world"
        }
    }
}```

**Review by baldawar — CHANGES_REQUESTED** [review]

**Comment by embano1** [comment]
> Just a heads up, this line triggered a failed build 

Hm, strange. Need to investigate with you if this persists (so far haven't seen this being reported from other users/CI and my own experience). Perhaps a Docker deamon version issue?

**Comment by embano1** [comment]
> 1.  "eventbridge-e2e" is too generic within config-json.json . For folks like me with hundreds of thousands of buses, its easy to hit conflict. I suggest renaming. 

Fair, but the default is for using the localstack e2e pipeline. If a user has a different setup, this can be overwritten as pointed out in the guide.

> why not connect-config.json instead of connnect-config-json.json ?

I had an avro config as well ;) but I agree it's confusing, changing.


2. why not `connect-config.json` instead of `connnect-config-json.json` ?

**Comment by embano1** [comment]
@baldawar comments answered/addressed. See here for diff: https://github.com/awslabs/eventbridge-kafka-connector/pull/77/commits/1f6edf8b5ffca9f8da43d6cc1a664df9b03902e4

Once you're ✅ I'll squash the commits for a final approval.

**Pull request merged**